src/helper/engine.cpp

```diff
@@ -58,12 +58,24 @@ void engine::update(engine_ruleset &ruleset,
     std::atomic_store(&common_, new_common);
 }
 
-std::optional<engine::result> engine::context::publish(parameter &&param)
+std::optional<engine::result> engine::context::publish(
+    parameter &&param, request_stage stage)
 {
     // Once the parameter reaches this function, it is guaranteed to be
     // owned by the engine.
     prev_published_params_.push_back(std::move(param));
 
+    std::optional<sampler::scope> opt_scope;
+    if (stage == request_stage::shutdown) {
```

Comment:
I think at this point you should also check if schema extraction is enabled and skip it if it isn't.

Reply:
The sampler checks for that

Reply:
Next line

Reply:
You missed the comment where I suggested you don't check in the sampler :-D

Reply:
Done